stdlib/public/Concurrency/TaskPrivate.h

```diff
@@ -561,12 +567,31 @@ struct AsyncTask::PrivateStorage {
 
     this->~PrivateStorage();
   }
+
+  swift::atomic<ActiveTaskStatus> &_status() {
+     return reinterpret_cast<swift::atomic<ActiveTaskStatus>&> (this->StatusStorage);
+  }
+
+  const swift::atomic<ActiveTaskStatus> &_status() const {
+    return reinterpret_cast<const swift::atomic<ActiveTaskStatus>&> (this->StatusStorage);
+  }
 };
 
-static_assert(sizeof(AsyncTask::PrivateStorage)
-                <= sizeof(AsyncTask::OpaquePrivateStorage) &&
-              alignof(AsyncTask::PrivateStorage)
-                <= alignof(AsyncTask::OpaquePrivateStorage),
+#if SWIFT_CONCURRENCY_ENABLE_PRIORITY_ESCALATION || SWIFT_POINTER_IS_8_BYTES
+// This means that ActiveTaskStatus is 128 bits wide - it needs to be 16 byte
+// aligned in AsyncTask
+static_assert(sizeof(ActiveTaskStatus) == sizeof(uint64_t) * 2);
+static_assert(((offsetof(AsyncTask, Private) + offsetof(AsyncTask::PrivateStorage, StatusStorage)) % 16 == 0),
```

Comment:
This is good, makes me a lot more comfortable with this wacky misalignment/realignment thing.